cmd/describe/machinepool/cmd.go

```diff
@@ -29,61 +30,68 @@ import (
-	Short:   "Show details of a machine pool on a cluster",
-	Long:    "Show details of a machine pool on a cluster.",
-	Example: `  # Show details of a machine pool named "mymachinepool"" on a cluster named "mycluster"
-  rosa describe machinepool --cluster=mycluster --machinepool=mymachinepool`,
-	Run:  run,
-	Args: cobra.MaximumNArgs(1),
-}
+const (
+	use     = "machinepool"
+	alias   = "machine-pool"
+	short   = "Show details of a machine pool on a cluster"
+	long    = "Show details of a machine pool on a cluster."
+	example = `  # Show details of a machine pool named "mymachinepool"" on a cluster named "mycluster"
+  rosa describe machinepool --cluster=mycluster --machinepool=mymachinepool`
+)
 
 var args struct {
 	machinePool string
 }
 
-func init() {
-	flags := Cmd.Flags()
-	flags.SortFlags = false
-	ocm.AddClusterFlag(Cmd)
-	output.AddFlag(Cmd)
+func NewDescribeMachinePoolCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     use,
+		Short:   short,
+		Long:    long,
+		Aliases: []string{alias},
+		Example: example,
+		Args:    cobra.NoArgs,
+		Run:     rosa.DefaultRunner(rosa.RuntimeWithOCM(), DescribeMachinePoolRunner()),
+	}
+
+	flags := cmd.Flags()
 	flags.StringVar(
 		&args.machinePool,
 		"machinepool",
 		"",
 		"Machine pool of the cluster to target",
 	)
-}
 
-func run(cmd *cobra.Command, argv []string) {
-	r := rosa.NewRuntime().WithAWS().WithOCM()
-	defer r.Cleanup()
-	err := runWithRuntime(r, cmd, argv)
-	if err != nil {
-		r.Reporter.Errorf(err.Error())
-		os.Exit(1)
-	}
+	output.AddFlag(cmd)
+	ocm.AddClusterFlag(cmd)
+	return cmd
 }
 
-func runWithRuntime(r *rosa.Runtime, cmd *cobra.Command, argv []string) error {
-	// Allow the use also directly the machine pool id as positional parameter
-	if len(argv) == 1 && !cmd.Flag("machinepool").Changed {
-		args.machinePool = argv[0]
-	}
-	if args.machinePool == "" {
-		return fmt.Errorf("You need to specify a machine pool name")
-	}
-	clusterKey := r.GetClusterKey()
-	cluster := r.FetchCluster()
-	if cluster.State() != cmv1.ClusterStateReady {
-		r.Reporter.Errorf("Cluster '%s' is not yet ready", clusterKey)
-		os.Exit(1)
-	}
-	isHypershift := cluster.Hypershift().Enabled()
+func DescribeMachinePoolRunner() rosa.CommandRunner {
+	return func(_ context.Context, runtime *rosa.Runtime, cmd *cobra.Command, argv []string) error {
+		// Allow the use also directly the machine pool id as positional parameter
+		if len(argv) == 1 && !cmd.Flag("machinepool").Changed {
+			args.machinePool = argv[0]
+		} else {
+			err := cmd.ParseFlags(argv)
+			if err != nil {
+				runtime.Reporter.Errorf("Unable to parse flags: %s", err)
+				os.Exit(1)
+			}
+		}
+		if args.machinePool == "" {
+			return fmt.Errorf("You need to specify a machine pool name")
+		}
+		clusterKey := runtime.GetClusterKey()
+		cluster := runtime.FetchCluster()
+		if cluster.State() != cmv1.ClusterStateReady {
+			runtime.Reporter.Errorf("Cluster '%s' is not yet ready", clusterKey)
+			os.Exit(1)
```

Comment:
@hunterkepley We can just return the `error` here and remove the `os.Exit(1)` call 👍